retry/retry.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+// IsRetriable is method signature for implementing to override default logic of retrying each error.
+type IsRetriable func(error) bool
+
+// Policy represents the struct to configure the retry behavior.
+type Policy struct {
+	// MaxAttempts represents the maximum number of retry attempts.
+	MaxAttempts int
+	// BackoffFactor is the multiplier by which retry interval (Jitter) increases after each retry.
+	// For constant backoff set Backoff factor to 1.
+	BackoffFactor float64
+	// Jitter is the interval before the first retry.
+	Jitter time.Duration
+	// ShouldRetry is the method to override default retry logic of retry every error.
```

Comment:
given the implementation of `isRetriable` function we should mention that if not specified in a Policy it will always retry or "default ShouldRetry always returns true".

Reply:
updated the comment for clarity.